### PR TITLE
[REVIEW] Removal of the old CertificateVerification implementation

### PR DESCRIPTION
**Pull request by NoelGraf**
- Removing the old CertificateVerification implementation and adapted the affected examples and tests.
- Rename UA_CertificateVerification_AcceptAll to UA_CertificateGroup_AcceptAll.
- Removal of the check_save_rejected_cert test file as the test is performed in the new check_certificategroup file.

### CODE REVIEW

**Comment by jpfr** [comment]
With the new API that takes only a single folder we should document very good how that folder has to look like.

That is still lacking…

We could even have an example with a pki folder in the repository.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1048,8 +1042,7 @@ list(APPEND plugin_sources
      ${PROJECT_SOURCE_DIR}/plugins/crypto/mbedtls/ua_securitypolicy_aes128sha256rsaoaep.c
      ${PROJECT_SOURCE_DIR}/plugins/crypto/mbedtls/ua_securitypolicy_aes256sha256rsapss.c
      ${PROJECT_SOURCE_DIR}/plugins/crypto/mbedtls/ua_mbedtls_create_certificate.c
-     ${PROJECT_SOURCE_DIR}/plugins/crypto/mbedtls/ua_certificategroup_mbedtls.c
-     ${PROJECT_SOURCE_DIR}/plugins/crypto/mbedtls/ua_certificategroup_memorystore_mbedtls.c)
```
Comment (mcpat) [inline CMakeLists.txt:1052]: Is there a reason why this is dropped?

File: plugins/crypto/mbedtls/ua_certificategroup_memorystore_mbedtls.c
Comment (mcpat) [inline plugins/crypto/mbedtls/ua_certificategroup_memorystore_mbedtls.c:1]: Same as above: why is the memory store deleted here?

Comment (mcpat) [reply plugins/crypto/mbedtls/ua_certificategroup_memorystore_mbedtls.c:1]: Ah ok, I got it: you merged it ino ua_certificategroup_mbedtls.c

File: CMakeLists.txt (hunk 1 above)
Comment (mcpat) [reply CMakeLists.txt:1052]: Should have checked the whole PR before asking ;-) - the sources are merged!

**Comment by NoelGraf** [comment]
@jpfr I have added additional documentation for the function and also detailed the folder structure.

**Comment by jpfr** [comment]
👍

**Pull request merged**